src/Pos/Update/Poll/Pure.hs

```diff
@@ -0,0 +1,118 @@
+-- | Pure Poll
+
+module Pos.Update.Poll.Pure
+       ( PurePoll (..)
+       , evalPurePollWithLogger
+       , execPurePollWithLogger
+       , runPurePollWithLogger
+       ) where
+
+import           Universum
+
+import           Control.Lens               (at, to, (%=), (.=))
+import qualified Data.HashMap.Strict        as HM
+import           System.Wlog                (CanLog, HasLoggerName (..), LogEvent,
+                                             NamedPureLogger, logDebug, logWarning,
+                                             runNamedPureLog)
+
+import           Pos.Crypto                 (hash)
+import           Pos.Types                  (SoftwareVersion (..))
+import           Pos.Update.Poll.Class      (MonadPoll (..), MonadPollRead (..))
+import qualified Pos.Update.Poll.PollState  as Poll
+import           Pos.Update.Poll.Types      (BlockVersionState (..),
+                                             DecidedProposalState (..),
+                                             UndecidedProposalState (..),
+                                             cpsSoftwareVersion, propStateToEither,
+                                             psProposal)
+
+newtype PurePoll a = PurePoll
+    { getPurePoll :: NamedPureLogger (State Poll.PollState) a
```

Comment:
This PR exists for 8 days already and there are still few things to do, I would like to merge it ASAP and then refine if necessary.